### ISSUE TITLE
Add `multiline_call_linter()`

### ISSUE THREAD

**Issue by AshesITR** [r-lib/lintr#1719]
Related but not the same as `indentation_linter()`, it would enforce the rules laid out in https://style.tidyverse.org/syntax.html#long-lines
It would produce three lints:

1. Closing parentheses of multi line function calls are on their own line.
2. Arguments on the first line of a multi line function call must be unnamed.
3. Arguments on subsequent lines of a multi line function call must be named.

It needs an `except=` argument to whitelist some functions, notably `paste()`, `paste0()`, `cat()`.
Please mention any additional functions you'd see fit for this.

Quoting examples from the styleguide:

```r
# Good
do_something_very_complicated(
  something = "that",
  requires = many,
  arguments = "some of which may be long"
)

map(x, f,
  extra_argument_a = 10,
  extra_argument_b = c(1, 43, 390, 210209)
)

paste0(
  "Requirement: ", requires, "\n",
  "Result: ", result, "\n"
)

# Bad
do_something_very_complicated("that", requires, many, arguments,
                              "some of which may be long"
                              )

not_paste0(
  "Requirement: ", requires, "\n",
  "Result: ", result, "\n"
)

paste0(
  "Requirement: ", requires,
  "\n", "Result: ",
  result, "\n")

```

**Comment by MichaelChirico** [comment]
plot() is another one where grouping keyword args helps a lot / putting
every argument on its own line harms readability. e.g. I always like to put
xlab= and xlim= or xaxt= on the same line

**Comment by dpprdan** [comment]
> Arguments on subsequent lines of a multi line function call must be named.

IMHO this should not be or only be implemented as a (non-default) `strict` option, since this is stricter than what the [tidyverse style-guide](https://style.tidyverse.org/syntax.html#long-lines) mentions.

> Short unnamed arguments __can__ also go on the same line as the function name, even if the whole function call spans multiple lines.

Emphasis mine. So this ought to be allowed:

```
map(
  my_list,
  some_function,
  extra_argument_a = 10,
  extra_argument_b = c(1, 43, 390, 210209)
)
```

([`map()`](https://purrr.tidyverse.org/reference/map.html) is not the best example for this anymore, since the tidyverse team recommends against passing arguments to `.f` via `...` now, but that doesn't really matter here).

**Comment by AshesITR** [comment]
> > Arguments on subsequent lines of a multi line function call must be named.
> 
> 
> 
> IMHO this should not be or only be implemented as a (non-default) `strict` option, since this is stricter than what the [tidyverse style-guide](https://style.tidyverse.org/syntax.html#long-lines) mentions.
> 
> 
> 
> > Short unnamed arguments __can__ also go on the same line as the function name, even if the whole function call spans multiple lines.
> 
> 
> 
> Emphasis mine. So this ought to be allowed:
> 
> 
> 
> ```
> 
> map(
> 
>   my_list,
> 
>   some_function,
> 
>   extra_argument_a = 10,
> 
>   extra_argument_b = c(1, 43, 390, 210209)
> 
> )
> 
> ```

I understand it to mean this style:

```r
map(my_list, some_function,
  extra_argument_a = 10,
  extra_argument_b = c(1, 43, 390, 210209)
)
```

**Comment by dpprdan** [comment]
Yes, both styles should be allowed.
